### ISSUE TITLE
The app does not respond to the close command

### ISSUE THREAD

**Issue by guglovich** [ValveSoftware/steam-for-linux#8468]
#### Your system information

* Steam client version (build number or date): 1646806235
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:

![Снимок экрана_2022-03-10_15-45-08](https://user-images.githubusercontent.com/54148165/157667758-ef63dccd-05a2-403c-a0fd-a74d437de6b9.png)
When you click to close Steam, the standard Steam close window pops up, but the client does not close. The Task Manager also does not show any processes or attempts to close. The closure can only be enforced. The problem also appears in the stable version of the client. Every time the client is killed and started, a new process descendant appears, which is the very 1st one in the screenshot above, and they pile up and pile up.
![Снимок экрана_2022-03-10_16-05-05](https://user-images.githubusercontent.com/54148165/157667943-a1e574b9-04b4-4994-b050-ee4f895ffb1d.png)

Second problem: After forcibly killing the main Steam processes, many child processes are left.

#### Steps for reproducing this issue:
Just try to close the client.

**Comment by kisak-valve** [comment]
Hello @guglovich, this reads like a similar issue to #6811 and #8113. Can you clarify exactly what sh is trying to run there? Your screenshot shows part of what was called with it being truncated in environment variables before the actual command.

**Comment by guglovich** [comment]
@kisak-valve I haven't figured out how to copy this from the task manager yet. This process, by the way, is not the only one that is duplicated, they also accumulate:

![Снимок экрана_2022-03-10_17-44-34](https://user-images.githubusercontent.com/54148165/157686112-fb60a8d5-6f2a-41d6-b271-0f603c38ab20.png)

![Снимок экрана_2022-03-10_17-44-59](https://user-images.githubusercontent.com/54148165/157686137-47f2620a-508d-45d5-a512-803ea91cf6d4.png)


![Снимок экрана_2022-03-10_17-44-46](https://user-images.githubusercontent.com/54148165/157686156-a3fca609-c509-477a-8c3e-1a0a05220065.png)

A little log from the terminal.

```
assert_20220310172353_38.dmp[29594]: Uploading dump (out-of-process)
/tmp/dumps/assert_20220310172353_38.dmp
src/clientdll/steamclient.cpp (854) : bufRet.TellPut() == sizeof(uint8)
src/clientdll/steamclient.cpp (854) : bufRet.TellPut() == sizeof(uint8)
assert_20220310172353_38.dmp[29594]: Finished uploading minidump (out-of-process): success = yes
assert_20220310172353_38.dmp[29594]: response: CrashID=bp-c3ee7cf3-053d-4df2-85d3-6244d2220310
assert_20220310172353_38.dmp[29594]: file ''/tmp/dumps/assert_20220310172353_38.dmp'', upload yes: ''CrashID=bp-c3ee7cf3-053d-4df2-85d3-6244d2220310''
Thread "CJobMgr::m_WorkThreadPool:2" (ID 29152) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:2" (ID 29152) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:2" (ID 29152) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:2" (ID 29152) failed to shut down
[2022-03-10 17:24:35] Background update loop checking for update. . .
[2022-03-10 17:24:35] Проверка на наличие обновлений...
[2022-03-10 17:24:35] Downloading manifest: https://cdn.akamai.steamstatic.com/client/steam_client_ubuntu12
Thread "CJobMgr::m_WorkThreadPool:2" (ID 29152) failed to shut down
```

**Comment by guglovich** [comment]
Thank you. That really turned out to be the reason.

**Comment by kisak-valve** [comment]
Thanks for testing. Closing as not a Steam client issue.

**Comment by guglovich** [comment]
I've found what the problem is. The changes in Proton-GE are taken from Proton Experimental. And this problem repeated itself when specifying Proton Experimental for the same game. Exactly the same processes start piling up and Steam won't close.

**Comment by kisak-valve** [comment]
https://github.com/ValveSoftware/Proton/issues/5680 is tracking the Proton Experimental variant of this issue.

**Comment by guglovich** [comment]
@kisak-valve The problem manifests itself even without Proton Experimental. A possible Proton 7.0 is also problematic.

**Comment by guglovich** [comment]
@kisak-valve I installed a git version of xfce4-taskmanager to copy the full paths of those processes.

`c:\windows\system32\steam.exe /home/perfectionist/.local/share/Steam/ubuntu12_32/../bin/d3ddriverquery64.exe                                                                                                                                                      `
`/Games/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier/pressure-vessel/bin/pv-bwrap --args 26 /usr/lib/pressure-vessel/from-host/bin/pressure-vessel-adverb --generate-locales --fd 19 --regenerate-ld.so-cache /run/pressure-vessel/ldso --add-ld.so-path /usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu --add-ld.so-path /usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu --set-ld-library-path /usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/aliases:/usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/aliases --exit-with-parent --subreaper --shell=none --terminal=none -- /Media/Games/SteamLibrary/steamapps/common/Proton 7.0/proton run /home/perfectionist/.local/share/Steam/ubuntu12_32/../bin/d3ddriverquery64.exe`

`python3 /Media/Games/SteamLibrary/steamapps/common/Proton 7.0/proton run /home/perfectionist/.local/share/Steam/ubuntu12_32/../bin/d3ddriverquery64.exe`

`sh -c LD_LIBRARY_PATH="$SYSTEM_LD_LIBRARY_PATH" PATH="$SYSTEM_PATH"  STEAM_COMPAT_APP_ID='1887720' STEAM_COMPAT_TOOL_PATHS='' STEAM_COMPAT_LIBRARY_PATHS='' STEAM_COMPAT_SHADER_PATH='/Media/Games/SteamLibrary/steamapps/shadercache/1887720' STEAM_COMPAT_INSTALL_PATH='/Media/Games/SteamLibrary/steamapps/common/Proton 7.0' STEAM_COMPAT_MOUNTS='' PRESSURE_VESSEL_SOCKET_DIR='/tmp/SteamPVSockets.2NMjPZ' STEAM_COMPAT_DATA_PATH='/Media/Games/SteamLibrary/steamapps/compatdata/1887720' STEAM_COMPAT_CLIENT_INSTALL_PATH='/home/perfectionist/.local/share/Steam' PROTON_CRASH_REPORT_DIR='/tmp/proton_crashreports' STEAM_FOSSILIZE_DUMP_PATH='/home/perfectionist/.local/share/Steam/shader_cache_temp_dir_d3d11_64/fozpipelinesv6/steamapprun_pipeline_cache' STEAM_FOSSILIZE_DUMP_PATH_READ_ONLY='$bucketdir/steam_pipeline_cache.foz;$bucketdir/steamapp_pipeline_cache.foz' FOSSILIZE_APPLICATION_INFO_FILTER_PATH='/home/perfectionist/.local/share/Steam/fossilize_engine_filters.json' ENABLE_VK_LAYER_VALVE_steam_fossilize_1='1' FONTCONFIG_`

**Comment by guglovich** [comment]
@kisak-valve I have fixed the problem. I don't know if this will work for others in the future but ask to do it just in case. In my case, removing mangohud completely helped. Installing mangohud even without running it immediately breaks wine and steam. 
Then after looking at AUR repositories I saw mangohud-nonvidia, installed it and no problems. Apparently there is a conflict between vanilla version and AMD hardware.
